components/brave_private_new_tab_ui/resources/page/hooks.ts

```diff
@@ -0,0 +1,39 @@
+import * as React from 'react'
+import getPageHandlerInstance from './api/brave_page_handler'
+
+export function useTorObserver () {
+  const [hasInit, setHasInit] = React.useState<boolean>(false)
+  const [isConnected, setIsConnected] = React.useState(false)
+  const [progress, setProgress] = React.useState<string | undefined>(undefined)
+
+  const isLoading = !hasInit && !isConnected
+
+  React.useEffect(() => {
+    getPageHandlerInstance().callbackRouter.onTorCircuitEstablished.addListener(setHasInit)
+    getPageHandlerInstance().callbackRouter.onTorInitializing.addListener(setProgress)
+  }, [])
+
+  React.useEffect(() => {
+    getPageHandlerInstance().pageHandler.getIsTorConnected()
+      .then((res: { isConnected: boolean }) => setIsConnected(res.isConnected))
```

Comment:
nit: you don't need to re-define type for `res` here (and other calls to similar functions in this module) since the type is coming from mojom's JSDoc. Might be a bit clearer to read without them.